tapchannel/aux_traffic_shaper.go

```diff
@@ -456,16 +457,43 @@ func (s *AuxTrafficShaper) ProduceHtlcExtraData(totalAmount lnwire.MilliSatoshi,
 		return totalAmount, htlcCustomRecords, nil
 	}
 
-	if htlc.RfqID.ValOpt().IsNone() {
-		return 0, nil, fmt.Errorf("no RFQ ID present in HTLC blob")
+	if htlc.AvailableRfqIDs.IsNone() {
+		return 0, nil, fmt.Errorf("no available RFQ IDs present in " +
+			"HTLC blob")
 	}
 
-	rfqID := htlc.RfqID.ValOpt().UnsafeFromSome()
 	acceptedQuotes := s.cfg.RfqManager.PeerAcceptedSellQuotes()
-	quote, ok := acceptedQuotes[rfqID.Scid()]
-	if !ok {
-		return 0, nil, fmt.Errorf("no accepted quote found for RFQ ID "+
-			"%x (SCID %d)", rfqID[:], rfqID.Scid())
+	availableIDs := htlc.AvailableRfqIDs.UnsafeFromSome().Val.IDs
+
+	var (
+		rfqID rfqmsg.ID
+		quote rfqmsg.SellAccept
+	)
+
+	// Let's find the quote that matches this peer. This will be the quote
+	// that we'll use to calculate the asset units. Given that we may only
+	// establish a maximum of 1 quote per peer, this check is safe to
```

Comment:
nit: to clarify my previous misunderstanding, change this to "establish a maximum of 1 quote per peer per payment, this check..."?